zkevm-circuits/src/evm_circuit/execution/push.rs

```diff
@@ -35,10 +40,33 @@ impl<F: Field> ExecutionGadget<F> for PushGadget<F> {
 
     fn configure(cb: &mut EVMConstraintBuilder<F>) -> Self {
         let opcode = cb.query_cell();
+        let code_size = cb.query_cell();
+        let code_size_left = code_size.expr() - cb.curr.state.program_counter.expr() - 1.expr();
 
         let value = cb.query_word32();
-        // Query selectors for each opcode_lookup
-        let selectors = array_init(|_| cb.query_bool());
+        // Query selectors for each opcode_lookup to know whether the current byte needs to be
+        // pushed
+        let is_pushed = array_init(|_| cb.query_bool());
+
+        // Query selectors for each opcode_lookup to know whether the current byte is actually
+        // padding
+        let is_padding = array_init(|_| cb.query_bool());
+
+        // Fetch the bytecode length from the bytecode table.
+        let code_hash = cb.curr.state.code_hash.clone();
+        cb.bytecode_length(code_hash.to_word(), code_size.expr());
```

Comment:
Suggestion: The spec uses `code_length` instead of `code_size`.  Since the lookup function is called `bytecode_length` I suggest updating all `code_size` to `code_length` to unify the variable names in the spec and here.